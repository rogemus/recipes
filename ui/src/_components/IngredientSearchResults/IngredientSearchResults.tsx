import type { FC } from "react";
import type { IngredientSearchResultsProps } from "./IngredientSearchResults.types";

const IngredientSearchResults: FC<IngredientSearchResultsProps> = ({
  ingredients = [],
  testId = "IngredientSearchResults",
}) => {
  if (ingredients.length === 0) return null;
  return (
    <ul data-testid={testId}>
      {ingredients.map(({ id, name }) => (
        <li key={id}>{name}</li>
      ))}
    </ul>
  );
};

export default IngredientSearchResults;
