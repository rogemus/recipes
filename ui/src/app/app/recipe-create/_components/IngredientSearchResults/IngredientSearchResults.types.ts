import type { Comp, IngredientSimple } from "@/_models";

export type IngredientSearchResultsProps = {
  ingredients: IngredientSimple[];
} & Comp;
