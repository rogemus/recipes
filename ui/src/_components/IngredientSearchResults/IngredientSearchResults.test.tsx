import { expect, test, describe } from "vitest";
import { render } from "@testing-library/react";

import IngredientSearchResults from "./IngredientSearchResults";

import type { IngredientSearchResultsProps } from "./IngredientSearchResults.types";

const defaultProps: IngredientSearchResultsProps = {
  ingredients: [
    {
      id: 5,
      name: "test ingredient",
    },
  ],
};

describe("@component/IngredientSearchResults", () => {
  test("renders component correctly", () => {
    const { container } = render(<IngredientSearchResults {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
