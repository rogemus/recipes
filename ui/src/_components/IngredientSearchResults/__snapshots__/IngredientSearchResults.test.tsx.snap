// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@component/IngredientSearchResults > renders component correctly 1`] = `
<div>
  <ul
    data-testid="IngredientSearchResults"
  >
    <li>
      test ingredient
    </li>
  </ul>
</div>
`;
